### PR TITLE
ELPP-1599 Cover PackagePOA

### PR DESCRIPTION
**Pull request by giorgiosironi**
This isn't support to work yet, but it helps documenting the process.

When running the test, I get as far as uploading the files to S3 and send a message with `econtools`:
```
2018-07-02T15:42:10Z INFO queue_workflow_starter_1659 1 message received
2018-07-02T15:42:10Z INFO queue_workflow_starter_1659 message contents: <boto.sqs.message.Message object at 0x7f1fb4c52390>
```
but then no workflow starts, we know some compatibility has to be tuned there.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
As far as I understand this is unrelated to https://github.com/elifesciences/elife-bot/pull/719 which would come into play later as the output of `PackagePOA`.

**Comment by giorgiosironi** [comment]
(can't yet run on Alfred as I am sorting out the new configuration file to be used there)

**Comment by giorgiosironi** [comment]
Managed to trigger the workflow using the filename rather than dates, it gets to:
```
2018-07-16T16:59:43Z INFO worker_1444 activityType: PackagePOA
2018-07-16T16:59:43Z INFO worker_1444 data: {
    "data": {
        "document": "50142_1_supp_mat_highwire_zip_853595_pxvg3m.zip"
    }
}
2018-07-16T16:59:52Z INFO worker_1444 Request to lax: GET http://end2end--gateway.elife.internal/articles/36157/versions
2018-07-16T16:59:52Z INFO worker_1444 Response from lax: 404
{"title": "not found"}
```

It has created 
```
$ aws s3 ls elife-poa-packaging-end2end/outbox/
2018-07-16 17:59:55   10896804 decap_elife_poa_e36157.pdf
2018-07-16 17:59:55       7688 elife_poa_e36157.xml
2018-07-16 17:59:55    3471149 elife_poa_e36157_ds.zip
```
which matches https://github.com/elifesciences/jira-import/issues/1196#issuecomment-402086465

So I'll now write some checks for these files, and try to use a random article id.

**Comment by gnott** [comment]
I think where it contacts Lax is if there is an existing pub date for the article it will use it. The 404 is fine, and if it gets a 404, it will use today's date as the pub date in the XML.

File: spectrum/checks.py [hunk 1]
```diff
@@ -783,20 +783,6 @@ def _get_host_ip(host):
 def _log_connection_error(e):
     LOGGER.debug("Connection error, will retry: %s", e)
 
-def _assert_status_code(response, expected_status_code, url):
```
Comment (giorgiosironi) [inline spectrum/checks.py]: moved this out to reduce the size of this module, getting bigger and bigger

**Comment by giorgiosironi** [comment]
Ready for review, but the test in unstable as often I get this output:
```
$ aws s3 ls elife-poa-packaging-end2end/outbox/ | grep 821034
2018-07-17 11:51:34   10896812 decap_elife_poa_e8210349340575836157.pdf
2018-07-17 11:51:34    3471513 elife_poa_e8210349340575836157_ds.zip
```
with the XML failing to be created. The linked PR will expand the logs but it's not the full story, how we discover what is happening?

**Comment by gnott** [comment]
The XML generation errors are usually emailed, like this

```
XML generation errors:
error_count: 1
error_messages: article_id 37798 error in set_articleType
```

The errors should also be in the ``/opt/elife-bot/xml_gen.log`` or the ``/opt/elife-bot/parse.log``, I cannot remember which. You can search for ``"error"``, it should say which of the 12 or 14 different attributes failed to set when generating XML.

**Comment by gnott** [comment]
Reviewing the code in the ``jats-generator`` library, I think the ``/opt/elife-bot/xml_gen.log`` is the best bet, look for ``"the following article did not have enough components and xml was not generated"``.

**Comment by giorgiosironi** [comment]
Main theory is that https://github.com/elifesciences/ejp-csv-parser/blob/develop/ejpcsvparser/csv_data.py#L189-L190 is caching the result of the CSV parsing after the first run, hence when the CSV files are updated the data about the new article inserted there would not appear. Could be a problem in production too if the workers are not restarted often enough?

**Comment by giorgiosironi** [comment]
To avoid further conflicts, going to resolve these ones and merging the test as skipped.

**Pull request merged**